plugins/queue/providers/kafka/lib/KafkaProvider.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+class KafkaProvider extends QueueProvider
+{
+	private $brokersArray;
+	private $server;
+	private $port;
+	private $conf;
+	protected $topic = null;
+	protected $producer = null;
+	
+	const DEFAULT_PORT = 29092;
+	const MAX_RETRIES = 3;
+	
+	public function __construct(array $kafkaConfig)
+	{
+		if ($kafkaConfig['brokers']) {
+			$brokers = $kafkaConfig['brokers'];
+			$brokersArray = explode(",", $brokers);
+			$newBrokersArray = array();
+			foreach ($brokersArray as &$broker) {
+				$hostAndPort = explode(":", $broker);
+				$host = $hostAndPort[0];
```

Comment:
I would do do something more like this
list($host, $port) = explode(":", $broker);
if(!$port)
{
	$port = self::DEFAULT_PORT;
}
$this->brokersArray[] = "$host:$port"